### PR TITLE
Feat: added platform to core vuln database

### PR DESCRIPTION
**Pull request by UlisesGascon**
### Main changes

As #912 enabled the field `affectedEnvironments` in to the core database, this PR will populate that information. 

I added to all the records the platform win, osx and linux unless there are specific reason to limit the scope as in 99 (046abcb93711c7a86ea41398f71906b564283742), 98 (86ed5d467109e5471824f86b26b536b57ca55d75) and 80 (950c20d3173e253399467237f39172e9e46bb155). 

I didn't check for specific affectedEnvironments to any vulnerabilty that is not affecting Node >=14.

### Context

- [x] #912 should be merged before as it is polluting the commits in this PR

### CODE REVIEW

File: tools/vuln_valid/vulnValidate.js [hunk 1]
```diff
@@ -31,7 +31,11 @@ const coreModel = joi.object().keys({
   type: joi.string().optional(),
   cvss_score: joi.number().optional(),
   cvss: joi.string().optional(),
-  reported_by: joi.string().optional()
+  reported_by: joi.string().optional(),
+  affectedEnvironments: joi
+  .array()
+  .items(joi.string().valid("mac", "win", "linux"))
```
Comment (RafaelGSS) [inline tools/vuln_valid/vulnValidate.js:37]: Same as https://github.com/nodejs/security-wg/pull/912#discussion_r1140255261

File: vuln/core/1.json [hunk 2]
```diff
@@ -6,5 +6,6 @@
   "vulnerable": "8.x || 7.x || 4.x || 6.x || 5.x",
   "patched": "^8.1.4 || ^7.10.1 || ^4.8.4 || ^6.11.1",
   "description": "memory overread when parsing invalid NAPTR responses",
-  "overview": "The c-ares function ares_parse_naptr_reply(), which is used for parsing NAPTR\nresponses, could be triggered to read memory outside of the given input buffer\nif the passed in DNS response packet was crafted in a particular way.\n\n"
+  "overview": "The c-ares function ares_parse_naptr_reply(), which is used for parsing NAPTR\nresponses, could be triggered to read memory outside of the given input buffer\nif the passed in DNS response packet was crafted in a particular way.\n\n",
+  "affectedEnvironments": ["linux", "osx", "win"]
```
Comment (RafaelGSS) [inline vuln/core/1.json:10]: I think we need a input for: 'all' or '*'

File: tools/vuln_valid/vulnValidate.js (hunk 1 above)
Comment (UlisesGascon) [reply tools/vuln_valid/vulnValidate.js:37]: yes, I created a branch from the #912 in order to do the validation before pushing the changes

Comment (UlisesGascon) [reply tools/vuln_valid/vulnValidate.js:37]: #912 should be merged before this one

File: vuln/core/1.json (hunk 2 above)
Comment (UlisesGascon) [reply vuln/core/1.json]: Solved in https://github.com/nodejs/security-wg/pull/912/commits/e3b8f8ae41e11601a45d08238d24d3ecfdfb3502 and [d9e544a](https://github.com/nodejs/security-wg/pull/914/commits/d9e544a102bfcf462f96fcdfb5d62ed701b663f1)

**Comment by UlisesGascon** [comment]
The pollution from #912 has gone and it is easier to review 👍

**Review by RafaelGSS — APPROVED** [review]

**Pull request merged**